bots/tests-invoke

```diff
@@ -297,9 +297,8 @@ class PullTask(object):
             os.environ["TEST_ATTACHMENTS"] = self.sink.attachments
 
         head = subprocess.check_output([ "git", "rev-parse", "HEAD" ], universal_newlines=True).strip()
-        if self.ref:
-            if not opts.offline:
-                subprocess.check_call([ "git", "fetch", self.remote, self.ref ])
+        if self.ref and not opts.offline:
+            subprocess.check_call([ "git", "fetch", self.remote, self.ref ])
```

Comment:
While this looks a bit nicer, it doesn't change behaviour at all (or am I blind?).

I'm afraid I don't actually comprehend the commit message; why would test-invoke be called "in a repository that isn't even checked out"?  -- After reading the next commit, I now understand it. But this still seems to be a no-op?